extensions/notebook/src/dialog/remoteBookDialog.ts

```diff
@@ -0,0 +1,335 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import * as loc from '../common/localizedConstants';
+import { RemoteBookController, IRelease, IAsset } from '../book/remoteBookController';
+import * as utils from '../common/utils';
+import { RemoteBook } from '../book/remoteBook';
+
+function apiGitHub(url: string): string {
+	return `https://api.github.com/${url}/releases`;
+}
+
+function getRemoteLocationCategory(name: string): azdata.CategoryValue {
+	if (name === loc.onGitHub) {
+		return { name: name, displayName: loc.onGitHub };
+	}
+	return { name: name, displayName: loc.onSharedFile };
+}
+
+export class RemoteBookDialogModel {
+	private _remoteLocation: string;
+	private _releases: IRelease[];
+	private _assets: IAsset[];
+	private _book: RemoteBook;
+
+	constructor() {
+	}
+
+	public get remoteLocation(): string {
+		return this._remoteLocation;
+	}
+
+	public set remoteLocation(location: string) {
+		this._remoteLocation = location;
+	}
+
+	public get releases(): IRelease[] {
+		return this._releases;
+	}
+
+	public set releases(newReleases: IRelease[]) {
+		this._releases = newReleases;
+	}
+
+	public get assets(): IAsset[] {
+		return this._assets;
+	}
+
+	public set assets(newAssets: IAsset[]) {
+		this._assets = newAssets;
+	}
+
+	public get remoteBook(): RemoteBook {
+		return this._book;
+	}
+
+	public set remoteBook(newBook: RemoteBook) {
+		this._book = newBook;
+	}
+
+}
+
+export class RemoteBookDialog {
+
+	private dialog: azdata.window.Dialog;
+	public view: azdata.ModelView;
+	private formModel: azdata.FormContainer;
+	private githubRepoDropdown: azdata.DropDownComponent;
+	private remoteLocationDropdown: azdata.DropDownComponent;
+	public releaseDropdown: azdata.DropDownComponent;
+	private searchButton: azdata.ButtonComponent;
+	public bookDropdown: azdata.DropDownComponent;
+	public versionDropdown: azdata.DropDownComponent;
+	public languageDropdown: azdata.DropDownComponent;
+	private _remoteTypes: azdata.CategoryValue[];
+	private readonly tigerToolboxRepo = 'repos/microsoft/tigertoolbox';
+	private readonly urlGithubRE = /^(?:https:\/\/(?:github\.com|api\.github\.com\/repos)|(?:\/)?(?:\/)?repos)([\w-.?!=&%*+:@\/]*)/g;
```

Comment:
tigetToolboxRepo and urlGithubRE can be defined outside the class - there's nothing object-specific about them.